bindings/python/pinocchio/visualize/panda3d_visualizer.py

```diff
@@ -0,0 +1,129 @@
+import multiprocessing
+import numpy as np
+import os
+import sys
+import uuid
+import warnings
+
+from pinocchio import libpinocchio_pywrap as pin
+from pinocchio.visualize.base_visualizer import BaseVisualizer
+from pinocchio.utils import npToTuple
+
+try:
+    import hppfcl
+    WITH_HPP_FCL_BINDINGS = True
+except:
+    WITH_HPP_FCL_BINDINGS = False
+
+from panda3d_viewer import Viewer as Panda3dViewer, ViewerClosedError
+
+__all__ = ['Panda3dVisualizer', 'Panda3dViewer', 'ViewerClosedError']
+
+
+class Panda3dVisualizer(BaseVisualizer):
+    """A Pinocchio display using panda3d engine."""
+    def initViewer(self, viewer=None, load_model=False):
+        """Init the viewer by attaching to / creating a GUI viewer."""
+        self.visual_group = None
+        self.collision_group = None
+        self.display_visuals = False
+        self.display_collisions = False
+        self.viewer = viewer
+
+        if viewer is None:
+            self.viewer = Panda3dViewer(window_title="python-pinocchio")
+
+        if load_model:
+            self.loadViewerModel()
+
+    def loadViewerModel(self, group_name='', color=None):
+        """Create the scene displaying the robot meshes in the viewer."""
+        # generate a unique group name if not specified
+        if not group_name:
+            group_name = str(uuid.uuid4())
```

Comment:
This makes the API of this viewer slightly different from the Gepetto viewer visualizer (I don't know for MeshCat). The original design was that, if you re-run a script, it doesn't duplicate the scene and reuse what has already been done. This random string will break this.

Do you have a strong motivation for this ?

Reply:
Hi @jmirabel , thanks for the comments! It is questionnable. My motivation is examples with  several robots, like this one: [examples/panda3d-viewer.py](https://github.com/stack-of-tasks/pinocchio/pull/1177/files#diff-c5a5325b610b9a6df51974e70d99f40c). Imagine if several robots have the same link name (e.g. right_arm_link_0).  In a default logic where all robots are attaching to a common root, a user who don't know about group_name parameter will get strange unobvious bugs: missing links, strange links movements.
I understand the default motivation for a Gepetto visualizer which is a remote and you have to cleanup previous scene each time, but Panda3D visualizer is a local one and starts with a clear scene.
An other option is to make group_name parameter required, to prevent users from getting confused.

Reply:
I see advantages to both a static default name and a random default one. I would follow those principles:
- the logic when you have several robots should appear where you have several robots.
- randomness prevents to know in advance what the name will be. This might be problematic.
- what happens if you re-create an instance of this viewer class in the same Python interpreter ?

I don't have a strong opinion, but I think the static name is better.

The default value may be attribute name of `Model`. This may be a *false good* (i.e. bad) idea because I don't know how often this is actually filled.

I don't mind it being required.

Reply:
@jmirabel what about using a hash of a model joints to avoid randomness:

```python
if not group_name:
    group_name = "model_%0X" % hash(
        (obj.name for obj in self.visual_model.geometryObjects))
```

It seems a good solution in all cases.

Reply:
:) I had not thought about that.

As it is very hard to predict, I would consider it as random (I agree it is repeatable, so better than random). I think I still prefer the name to be required but I won't fight too much and let others express their opinion.

Reply:
OK, the group_name is a required parameter now.